src/Catty/DataModel/Bricks/Sound/PlaySoundAndWaitBrick.m

```diff
@@ -0,0 +1,94 @@
+/**
+ *  Copyright (C) 2010-2019 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+#import "PlaySoundAndWaitBrick.h"
+#import "Sound.h"
+#import "CBMutableCopyContext.h"
+
+@implementation PlaySoundAndWaitBrick
+
+- (kBrickCategoryType)category
+{
+    return kSoundBrick;
+}
+
+#pragma mark - Copy
+- (id)mutableCopyWithContext:(CBMutableCopyContext*)context
+{
+    if (! context) NSError(@"%@ must not be nil!", [CBMutableCopyContext class]);
+    PlaySoundAndWaitBrick *brick = [[self class] new];
+    
+    id updatedReference = [context updatedReferenceForReference: self.sound];
+    
+    if (updatedReference != nil) {
+        brick.sound = updatedReference;
+    } else {
+        brick.sound = self.sound;
+    }
+    
+    return brick;
+}
+
+#pragma mark - Description
+- (NSString*)description
+{
+    return [NSString stringWithFormat:@"PlaySoundAndWait (File Name: %@)", self.sound.fileName];
+}
+
+- (BOOL)isEqualToBrick:(Brick*)brick
+{
+    if(![self.sound isEqualToSound:((PlaySoundAndWaitBrick*)brick).sound])
```

Comment:
Could you please double check this line. What happens if the Brick is not of type `PlaySoundAndWaitBrick`?

Reply:
Thanks, I'll add a type check to this method.

Reply:
Perfect, thanks!